interfaces/cython/cantera/examples/reactors/ic_engine.py

```diff
@@ -54,200 +56,203 @@
+ax[1].plot(t, states.T)
+ax[1].set_ylabel('$T$ [K]')
+ax[1].set_xlabel('$\phi$ [deg]')
+ax[1].set_xticklabels(ca_ticks(ax[1].get_xticks()))
+fig.savefig('ic_engine_t_p_T.png')
 plt.show()
-plt.savefig('ic_engine_t_p_T.png')
 
 # p-V diagram
-plt.clf()
-plt.plot(V[t > 0.04] * 1000, states.P[t > 0.04] / 1.e5)
-plt.xlabel('$V$ [l]')
-plt.ylabel('$p$ [bar]')
+fig, ax = plt.subplots()
+ax.plot(states.V[t > 0.04] * 1000, states.P[t > 0.04] / 1.e5)
+ax.set_xlabel('$V$ [l]')
+ax.set_ylabel('$p$ [bar]')
+fig.savefig('ic_engine_p_V.png')
 plt.show()
-plt.savefig('ic_engine_p_V.png')
 
 # T-S diagram
-plt.clf()
-plt.plot(m[t > 0.04] * states.s[t > 0.04], states.T[t > 0.04])
-plt.xlabel('$S$ [J/K]')
-plt.ylabel('$T$ [K]')
+fig, ax = plt.subplots()
+ax.plot(states.m[t > 0.04] * states.s[t > 0.04], states.T[t > 0.04])
+ax.set_xlabel('$S$ [J/K]')
+ax.set_ylabel('$T$ [K]')
+fig.savefig('ic_engine_T_S.png')
 plt.show()
-plt.savefig('ic_engine_T_S.png')
 
 # heat of reaction and expansion work
-plt.clf()
-plt.plot(t, heat_release_rate, label='$\dot{Q}$')
-plt.plot(t, d_W_v_d_t, label='$\dot{W}_v$')
-plt.ylim(-1e5, 1e6)
-plt.legend(loc=0)
-plt.ylabel('[W]')
-plt.xlabel('$\phi$ [deg]')
-plt.xticks(plt.xticks()[0], crank_angle(plt.xticks()[0]) * 180 / np.pi,
-           rotation=17)
+fig, ax = plt.subplots()
+ax.plot(t, 1.e-3 * states.heat_release_rate, label='$\dot{Q}$')
+ax.plot(t, 1.e-3 * states.dWv_dt, label='$\dot{W}_v$')
+ax.set_ylim(-1e2, 1e3)
+ax.legend(loc=0)
+ax.set_ylabel('[kW]')
+ax.set_xlabel('$\phi$ [deg]')
+ax.set_xticklabels(ca_ticks(ax.get_xticks()))
+fig.savefig('ic_engine_Q_W.png')
 plt.show()
-plt.savefig('ic_engine_Q_W.png')
 
 # gas composition
-plt.clf()
-plt.plot(t, states('O2').X, label='O2')
-plt.plot(t, states('CO2').X, label='CO2')
-plt.plot(t, states('CO').X, label='CO')
-plt.plot(t, states('C3H8').X * 10, label='C3H8 x10')
-plt.legend(loc=0)
-plt.ylabel('$X_i$ [-]')
-plt.xlabel('$\phi$ [deg]')
-plt.xticks(plt.xticks()[0], crank_angle(plt.xticks()[0]) * 180 / np.pi,
-           rotation=17)
+fig, ax = plt.subplots()
+ax.plot(t, states('O2').X, label='O2')
+ax.plot(t, states('CO2').X, label='CO2')
+ax.plot(t, states('CO').X, label='CO')
+ax.plot(t, states('C3H8').X * 10, label='C3H8 x10')
+ax.legend(loc=0)
+ax.set_ylabel('$X_i$ [-]')
+ax.set_xlabel('$\phi$ [deg]')
+ax.set_xticklabels(ca_ticks(ax.get_xticks()))
+fig.savefig('ic_engine_t_X.png')
 plt.show()
-plt.savefig('ic_engine_t_X.png')
 
 
-#####################################################################
+# ####################################################################
 # Integral Results
-#####################################################################
+# ####################################################################
 
-from scipy.integrate import trapz
-Q = trapz(heat_release_rate, t)
-W = trapz(d_W_v_d_t, t)
+
+# heat release
+Q = trapz(states.heat_release_rate, t)
+print('{:45s}{:>4.1f} kW'.format('Heat release rate per cylinder (estimate):',
```

Comment:
I think you used the `{:45s}` to space these strings all the same, but it looks weird to me to have a literal string as an argument to `format()` 🤷‍♂

Reply:
That was indeed the rationale (replaced a bunch of `\t`'s. Better idea?